Include/pythread.h

```diff
@@ -20,7 +20,7 @@ PyAPI_FUNC(unsigned long) PyThread_start_new_thread(void (*)(void *), void *);
 PyAPI_FUNC(void) _Py_NO_RETURN PyThread_exit_thread(void);
 PyAPI_FUNC(unsigned long) PyThread_get_thread_ident(void);
 
-#if defined(__APPLE__) || defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(_WIN32) || defined(_AIX)
+#if defined(__APPLE__) || defined(__linux__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || defined(__DragonFly__) || defined(_WIN32) || defined(_AIX)
```

Comment:
Ugh, we should break this up. @vstinner, are you ok with applying the following suggestion to this PR?

```suggestion
#if defined(__APPLE__)          \
    || defined(__linux__)       \
    || defined(__FreeBSD__)     \
    || defined(__OpenBSD__)     \
    || defined(__NetBSD__)      \
    || defined(__DragonFly__)   \
    || defined(_WIN32)          \
    || defined(_AIX)
#define PY_HAVE_THREAD_NATIVE_ID 
```

Reply:
Broken up by 5a54ad242fee3a3833c8478e51c2232328f6230a.